### PR TITLE
core: fix T5 statediffs

### PR DESCRIPTION
**Pull request by AnnaShaleva**
This PR makes us compatible up to 187157 block of T5.

### CODE REVIEW

File: pkg/core/interop/context.go [hunk 1]
```diff
@@ -207,6 +207,9 @@ func (c *ContractMD) AddMethod(md *MethodAndPrice, desc *manifest.Method) {
 	index := sort.Search(len(c.Manifest.ABI.Methods), func(i int) bool {
 		md := c.Manifest.ABI.Methods[i]
 		if md.Name != desc.Name {
+			if md.Name == "getCandidateVote" && desc.Name == "getCandidates" {
```
Comment (roman-khimov) [inline pkg/core/interop/context.go:210]: Such a kludge... How about comparing `strings.ToLower()` values? This method is not on the hot path.

File: pkg/core/native/management.go [hunk 2]
```diff
@@ -615,7 +615,7 @@ func checkScriptAndMethods(script []byte, methods []manifest.Method) error {
 	offsets := bitfield.New(l)
 	for i := range methods {
 		if methods[i].Offset >= l {
-			return errors.New("out of bounds method offset")
+			continue
```
Comment (roman-khimov) [inline pkg/core/native/management.go:618]: C# node bug, please.

Comment (roman-khimov) [reply pkg/core/native/management.go:618]: And btw, I think it'll be compatible with `methods[i].Offset > l`.

Comment (AnnaShaleva) [reply pkg/core/native/management.go:618]: > C# node bug, please.

https://github.com/neo-project/neo/issues/2767. But technically, such script can be safely handled by both C# and Go VMs, see https://github.com/nspcc-dev/neo-go/blob/80823c58d96a8a2542df1c1e92b57f7af7f67e24/pkg/vm/context.go#L108.

>  it'll be compatible with `methods[i].Offset > l`

Why? Only for the current case, but for other possible cases it won't, see https://github.com/neo-project/neo-vm/blob/a65487fa56be3eccb2c1dbfec5dcdd71b8a05fde/src/Neo.VM/Script.cs#L146.

Comment (roman-khimov) [reply pkg/core/native/management.go:618]: I keep underestimating C# node.

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
Some test if failing.

**Comment by AnnaShaleva** [comment]
Wow, the first commit had a bad bug, likely we have `TestNativeHelpersCompile`. Fixed.

File: pkg/core/interop/context.go [hunk 3]
```diff
@@ -236,7 +236,7 @@ func (c *ContractMD) GetMethodByOffset(offset int) (MethodAndPrice, bool) {
 func (c *ContractMD) GetMethod(name string, paramCount int) (MethodAndPrice, bool) {
 	index := sort.Search(len(c.Methods), func(i int) bool {
 		md := c.Methods[i]
-		res := strings.Compare(name, md.MD.Name)
+		res := strings.Compare(strings.ToLower(name), strings.ToLower(md.MD.Name))
```
Comment (roman-khimov) [inline pkg/core/interop/context.go:239]: Will affect every call, meh... OK, whatever, let's make it compatible for now.

Comment (AnnaShaleva) [reply pkg/core/interop/context.go:239]: It is used for Oracle contract only, so it's not critical.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**